spec/logstash/circuit_breaker_spec.rb

```diff
@@ -14,33 +14,55 @@ class DummyErrorTest < StandardError; end
 
   subject { LogStash::CircuitBreaker.new("testing", options) }
 
+  context "when the breaker is closed" do
+    it "closed by default" do
+      expect(subject.closed?).to eq(true)
+    end
 
-  it "closed by default" do
-    expect(subject.closed?).to eq(true)
-  end
-
-  context "when having too many errors" do
-    let(:future_time) { Time.now + 3600 }
-    before do
-      subject.execute do
-        raise DummyErrorTest
-      end
+    it "always raise an exception if an errors occur" do
+      expect {
+        subject.execute do
+          raise DummyErrorTest
+        end
+      }.to raise_error(LogStash::CircuitBreaker::HalfOpenBreaker)
     end
 
-    it "raised an exception if we have too many errors" do
+    it "open if we pass the errors threadshold" do
+      expect {
+        subject.execute do
+          raise DummyErrorTest
+        end
+      }.to raise_error(LogStash::CircuitBreaker::HalfOpenBreaker)
+
       expect {
         subject.execute do
           raise DummyErrorTest
         end
       }.to raise_error(LogStash::CircuitBreaker::OpenBreaker)
     end
+  end
+
+  context "When the breaker is open" do
+    let(:future_time) { Time.now + 3600 }
+
+    before do
+      # trip the breaker
+      (error_threshold + 1).times do
+        begin
+          subject.execute do
+            raise DummyErrorTest
+          end
+        rescue
+        end
+      end
+    end
 
-    it "sets the breaker to open" do
+    it "#closed? should return false" do
       expect(subject.closed?).to eq(false)
     end
 
     it "resets the breaker after the time before retry" do
-      expect(Time).to receive(:now).at_least(1).and_return(future_time)
+      expect(Time).to receive(:now).at_least(2).and_return(future_time)
```

Comment:
![delorean-back-to-the-future](https://cloud.githubusercontent.com/assets/640/9281083/bb9031c8-4290-11e5-9489-b37a95c86a91.jpg)